drivers/QSPI.cpp

```diff
@@ -0,0 +1,271 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "drivers/QSPI.h"
+#include "platform/mbed_critical.h"
+
+#if DEVICE_QSPI
+
+#define IS_BUS_WIDTH_VALID(width)          ((width == QSPI_CFG_BUS_SINGLE) || (width == QSPI_CFG_BUS_DUAL) || (width == QSPI_CFG_BUS_QUAD))    
```

Comment:
Besides that these can be small function helpers that can  be inlined and of course in the unnamed namespace or just static, will be easier if this preconditions are visible for a user. The main question is do we need these checks? If bus width is type of alt_width, it can be in range of valid values.

Reply:
I think you are right, we dont need the check there as we are exposing the enums defs from HAL upto App. We can optimize those checks.